### DISCUSSION TITLE
why usecase are using flow()

### DISCUSSION THREAD

**Discussion by periva101** [serbelga/Todometer-KMP#142] [General]
I want to thank you first,
then I hope you answer my question, it is a question, not an issue 😃

All ApiService methods are returning suspend functions
` operator fun invoke(dadJoke: DadJoke, favored: Boolean): Flow<Boolean>
`
why are FavorDadJokeInteractor methods converted to` flowOf ()`
e.g
```
 override fun invoke(dadJoke: DadJoke, favored: Boolean): Flow<Boolean> {
        return flow {
            val updates = repository.favorDadJoke(
                id = dadJoke.id,
                favored = favored
            )
            emit(updates > 0)
        }
    }
```
what are the benefits of making all repo methods in kotlin flow,
where consuming suspend fun is easier than flow?
also combing two suspend funs is much easier to combine two flows to create new use cases.

I see this pattern I would like to know the secret behind 😎

**Comment by periva101** [comment]
@serbelga

**Comment by serbelga** [comment]
I'm moving this question to **Discussions** because it makes more sense to be here instead of **Issues**. First, when you make HTTP requests to an API, these are one-shot read / write operations. It is not about observing a data stream in a reactive way. This is why when you define the skeleton of your API calls you do so using coroutines, and in your repository you use suspend functions.
On the other hand, if you use a local database in Room for example, which offers the possibility to observe changes in your database, you can use Flow to observe the data stream. In that case, if you make a call to the local data source in your repository, you would use Flow in the repository. That means that you would use suspend functions or flow depending of your needs.

Here, there's more info about this: https://developer.android.com/topic/architecture/data-layer#expose-apis

**Reply by periva101** [reply]
@serbelga , sorry 😢 I do not that about GitHub Discussions, and thank a lot for the answer , I appreciated it 😊
